### PR TITLE
Fix abstract types not being handled when nested in object types

### PR DESCRIPTION
**Pull request by eddeee888**
## Description

This PR fixes a scenario where abstract types are not pointing to generated resolvers type correctly:
- an object type A...
- with a field that returns an object type B (could be self-referencing to A) that has an abstract type field.

In said case, the object type B must use generated resolver types, otherwise, it may have type conflicts

Related https://github.com/dotansimha/graphql-code-generator/issues/10004

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit test
- [x] Integration test

### CODE REVIEW

File: packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts [hunk 1]
```diff
@@ -1835,3 +1854,84 @@ function normalizeResolversNonOptionalTypename(
     ...input,
   };
 }
+
+function checkIfObjectTypeHasAbstractTypesRecursively(
+  baseType: GraphQLObjectType,
+  result: {
+    isObjectWithAbstractType: boolean;
+    checkedTypesWithNestedAbstractTypes: Record<string, { hasNestedAbstractTypes: 'yes' | 'no' | 'checking' }>;
```
Comment (dimaMachina) [inline packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts:1862]: can it have a better name ? for me, anything which starts with `has*` should be a boolean

**Review by dotansimha — APPROVED** [review]

File: packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts (hunk 1 above)
Comment (eddeee888) [reply packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts:1862]: True! It started as `boolean` but then I saw the need to change to 3 different options, but didn't review the name.

Do you have suggestions? I'm thinking a few options but I'm unsure about them:
- `value`
- `result`
- `checkStatus`

File: packages/plugins/typescript/resolvers/tests/ts-resolvers.spec.ts
Comment (eddeee888) [inline packages/plugins/typescript/resolvers/tests/ts-resolvers.spec.ts:1]: `resolversNonOptionalTypename` tests are moved to `ts-resolvers.config.ResolversNonOptionalTypename` to further split tests

File: packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts (hunk 1 above)
Comment (dimaMachina) [reply packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts:1862]: checkStatus For me is good

**Comment by dylanwulf** [comment]
@eddeee888 Hi, are there any config options to revert this behavior? On `@graphql-codegen/typescript-resolvers` v4.2.1 I am getting an error during tsc type check which did not happen in v4.2.0:
`RangeError: Maximum call stack size exceeded`

**Comment by eddeee888** [comment]
Hi @dylanwulf, could you please open an issue here with an example codegen config + snippet of your schema? https://github.com/dotansimha/graphql-code-generator/issues/new?assignees=&labels=&projects=&template=bug_report.yml?

**Comment by dylanwulf** [comment]
> Hi @dylanwulf, I think the first step is to understand the issue. Could you please open an issue here with an example codegen config + snippet of your schema? https://github.com/dotansimha/graphql-code-generator/issues/new?assignees=&labels=&projects=&template=bug_report.yml

No problem, issue opened here: #10084

**Pull request merged**